yt/frontends/amrvac/data_structures.py

```diff
@@ -250,15 +251,110 @@ def _set_code_unit_attributes(self):
         # This is the reason why it uses setdefaultattr: it will only fill the gaps lefts by the "override",
         # instead of overriding them again
 
-        setdefaultattr(self, "length_unit", self.quan(1, "cm"))
-        setdefaultattr(self, "mass_unit", self.quan(1, "g"))
-        setdefaultattr(self, "time_unit", self.quan(1, "s"))
+        amrvac_units_dict = self._get_amrvac_normalisations()
+
+        setdefaultattr(self, "length_unit", amrvac_units_dict['length_unit'])
+        setdefaultattr(self, "mass_unit", amrvac_units_dict['mass_unit'])
+        setdefaultattr(self, "time_unit", amrvac_units_dict['time_unit'])
+
+        setdefaultattr(self, "velocity_unit", amrvac_units_dict['velocity_unit'])
+        setdefaultattr(self, "density_unit", amrvac_units_dict['density_unit'])
+        setdefaultattr(self, "numberdensity_unit", amrvac_units_dict['numberdensity_unit'])
+
+        setdefaultattr(self, "temperature_unit", amrvac_units_dict['temperature_unit'])
+        setdefaultattr(self, "pressure_unit", amrvac_units_dict['pressure_unit'])
+        setdefaultattr(self, "magnetic_unit", amrvac_units_dict['magnetic_unit'])
+
+
+    def _get_amrvac_normalisations(self):
```

Comment:
after carefully reading this whole function, I don't see a good reason not to merge it with `_set_code_unit_attributes`.